rasa_sdk/grpc_server.py

```diff
@@ -0,0 +1,248 @@
+from __future__ import annotations
+
+import signal
+
+import asyncio
+
+import grpc
+import logging
+import types
+from typing import Union, Optional
+from concurrent import futures
+from grpc import aio
+from google.protobuf.json_format import MessageToDict, ParseDict
+
+from rasa_sdk.constants import (
+    DEFAULT_SERVER_PORT,
+    DEFAULT_ENDPOINTS_PATH,
+    NO_GRACE_PERIOD,
+)
+from rasa_sdk.executor import ActionExecutor
+from rasa_sdk.grpc_errors import (
+    ResourceNotFound,
+    ResourceNotFoundType,
+    ActionExecutionFailed,
+)
+from rasa_sdk.grpc_py import (
+    action_webhook_pb2,
+    action_webhook_pb2_grpc,
+    health_pb2_grpc,
+)
+from rasa_sdk.grpc_py.action_webhook_pb2 import (
+    ActionsResponse,
+    ActionsRequest,
+    WebhookRequest,
+)
+from rasa_sdk.grpc_py.health_pb2 import HealthCheckRequest, HealthCheckResponse
+from rasa_sdk.interfaces import (
+    ActionExecutionRejection,
+    ActionNotFoundException,
+    ActionMissingDomainException,
+)
+from rasa_sdk.tracing.utils import (
+    get_tracer_and_context,
+    TracerProvider,
+)
+from rasa_sdk.utils import (
+    check_version_compatibility,
+    number_of_sanic_workers,
+    file_as_bytes,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class GRPCActionServerHealthCheck(health_pb2_grpc.HealthServiceServicer):
+    """Runs health check RPC which is served through gRPC server."""
+
+    def __init__(self) -> None:
+        """Initializes the HealthServicer."""
+        pass
+
+    def Check(self, request: HealthCheckRequest, context) -> HealthCheckResponse:
+        """Handle RPC request for the health check.
+
+        Args:
+            request: The health check request.
+            context: The context of the request.
+
+        Returns:
+            gRPC response.
+        """
+        response = HealthCheckResponse()
+        return response
+
+
+class GRPCActionServerWebhook(action_webhook_pb2_grpc.ActionServiceServicer):
+    """Runs webhook RPC which is served through gRPC server."""
+
+    def __init__(
+        self,
+        executor: ActionExecutor,
+        tracer_provider: Optional[TracerProvider] = None,
+    ) -> None:
+        """Initializes the ActionServerWebhook.
+
+        Args:
+            tracer_provider: The tracer provider.
+            executor: The action executor.
+        """
+        self.tracer_provider = tracer_provider
+        self.executor = executor
+
+    async def Actions(self, request: ActionsRequest, context) -> ActionsResponse:
```

Comment:
Same question here about not using the method args as above. Also context is missing type hint.

Reply:
I tried but I could not find the type of the context anywhere.

Considering method args, this is the cleanest way to do it according to [gRPC docs](https://protobuf.dev/programming-guides/api/#unique-protos). 
It provides good options for backwards compatibility later on.

Reply:
Found the type of context after some digging through docs. It's `grpc.aio.ServicerContext`.

Reply:
Nice 💯 could we add it in?